deploy/cloudformation/manifest-pipeline-pipeline.yml

```diff
@@ -0,0 +1,769 @@
+      Description: !Sub "Building stage for ${AppName}."
+      Environment:
+        ComputeType: BUILD_GENERAL1_SMALL
+        EnvironmentVariables:
+          - Name: S3_BUCKET
+            Value: !Ref S3Bucket
+        Image: !Ref CodeBuildImage
+        Type: LINUX_CONTAINER
+      Name: !Sub "${AppName}-build"
+      ServiceRole: !GetAtt
+        - CodeBuildTrustRole
+        - Arn
+      Source:
+        Type: CODEPIPELINE
+      Tags:
+        - Key: app-name
+          Value: !Ref AppName
+      TimeoutInMinutes: 5
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec: !Sub |
+          version: 0.2
+          phases:
+            install:
+              commands:
+                - echo "Ensure that the codebuild directory is executable"
+                - chmod -R 755 ./scripts/codebuild/*
+                - ./scripts/codebuild/install.sh
+            pre_build:
+              commands:
+                - ./scripts/codebuild/pre_build.sh
+            build:
+              commands:
+                - ./scripts/codebuild/build.sh
+            post_build:
+              commands:
+                - ./scripts/codebuild/post_build.sh
+          artifacts:
+            files:
+              - output.yml
+              - test-stack-configuration.json
+              - prod-stack-configuration.json
+
+  S3Bucket:
+    Description: Creating Amazon S3 bucket for AWS CodePipeline artifacts
+    Type: AWS::S3::Bucket
+    DeletionPolicy: Retain
+    Properties:
+      VersioningConfiguration:
+        Status: Enabled
+
+  S3ArtifactBucketPolicy:
+    DependsOn: [S3Bucket]
+    Description: Setting Amazon S3 bucket policy for AWS CodePipeline access
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref S3Bucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Id: SSEAndSSLPolicy
+        Statement:
+        - Sid: DenyInsecureConnections
+          Effect: Deny
+          Principal: "*"
+          Action: s3:*
+          Resource: !Sub "arn:aws:s3:::${S3Bucket}/*"
+          Condition:
+            Bool:
+              aws:SecureTransport: false
+
+  ProjectPipeline:
+    DependsOn: [S3Bucket, CodeBuildProject]
+    Description: Creating a deployment pipeline for your project in AWS CodePipeline
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      Name: !Sub "${AppName}-pipeline"
+      RoleArn: !GetAtt
+        - CodePipelineTrustRole
+        - Arn
+      Stages:
+      - Name: Source
+        Actions:
+        - Name: SourceFromRepo
+          InputArtifacts: []
+          ActionTypeId:
+            Version: "1"
+            Category: Source
+            Owner: ThirdParty
+            Provider: GitHub
+          OutputArtifacts:
+          - Name: SourceFromRepo
+          Configuration:
+            Repo: !Ref GitHubRepoName
+            Branch: !Ref GitHubRepoBranch
+            OAuthToken: !Ref GitHubToken
+            Owner: !Ref GitHubUser
+          RunOrder: 1
+        -
```

Comment:
I'm not sure you need to monitor the Config repository - this would also allow you to clean up some of the parameters. Can you show me where you're using this code in the Pipeline or CodeBuilds?

Reply:
So if you see there is a line in the buildspec that says 

- ./scripts/codebuild/build.sh

That script is in the code repo.  https://github.com/ndlib/mellon-manifest-pipeline/blob/0a18bbcc9091d8fe3d570878b3ca40d9b7bbbad8/scripts/codebuild/build.sh#L7

That line copies the cloudformation for the lambda application into the code repo so that it can be packaged.  

The reason that the files in script/codebuild exist is so that i can run them in the buildspec and also run them in a local deploy / setup way.  

See the install.sh file.   https://github.com/ndlib/mellon-manifest-pipeline/blob/0a18bbcc9091d8fe3d570878b3ca40d9b7bbbad8/scripts/codebuild/install.sh

This does a very specific install that (while it currently does not work) can be used to setup a local file set and one that can run in the lambda.